src/opers.c

```diff
@@ -1981,2769 +1952,627 @@ static inline Obj CacheOper (
-      res = CALL_2ARGS( method, arg1, arg2 );
-    }
-    while (res == TRY_NEXT_METHOD );
-
-    /* return the result                                                   */
-    return res;
-}
-
-
-
-/****************************************************************************
-**
-**  DoConstructor3Args( <oper>, <a1>, <a2>, <a3> )
-*/
-Obj DoConstructor3Args (
-    Obj                 oper,
-    Obj                 arg1,
-    Obj                 arg2,
-    Obj                 arg3 )
-{
-    Obj                 res;
-    Obj                 type1;
-    Obj                 type2;
-    Obj                 id2;
-    Obj                 type3;
-    Obj                 id3;
-    Obj *               cache;
-    Obj                 method;
-    Int                 i;
-    Obj                 prec;
-
-    /* get the types of the arguments                                      */
-    while (!IS_OPERATION(arg1))
-      {
-        arg1 = ErrorReturnObj(
-                "Constructor: the first argument must be a filter not a %s",
-                (Int)TNAM_OBJ(arg1), 0L, 
-                "you can replace the first argument <arg1> via 'return <arg1>;'");
-      }
-    type1 = FLAGS_FILT( arg1 );
-    type2 = TYPE_OBJ_FEO( arg2 );
-    id2 = ID_TYPE( type2 );
-    type3 = TYPE_OBJ_FEO( arg3 );
-    id3 = ID_TYPE( type3 );
-
-    /* try to find an applicable method in the cache                       */
-    prec = INTOBJ_INT(-1);
-
-    do {
-      /* The next line depends on the implementation of INTOBJS */
-      prec = (Obj)(((Int)prec) +4);
-      method = 0;
-
-      /* recalculate cache each pass, in case of GC     */
-      cache = 1+ADDR_OBJ( CacheOper( oper, 3 ) );
-
-      /* Up to CACHE_SIZE methods might be in the cache */
-      if (prec < INTOBJ_INT(CACHE_SIZE))
-        {
-          for (i = 0;  i < 5*CACHE_SIZE; i+= 5) {
-            if (  cache[i+1] == prec && cache[i+2] == type1
-                  && cache[i+3] == id2 && cache[i+4] == id3 ) {
-              method = cache[i];
-#ifdef COUNT_OPERS
-              OperationHit++;
-#endif
-              break;
-            }
-          }
-        }
-      
-      /* otherwise try to find one in the list of methods                    */
-      if (!method)
-        {
-          if (prec == INTOBJ_INT(0))
-            method = CALL_4ARGS( Constructor3Args, oper, type1, type2, type3 );
-          else
-            method = CALL_5ARGS( NextConstructor3Args, oper, prec, type1, type2, type3 );
-          
-          while (method == Fail)
-            {
-              Obj arglist[3];
-              arglist[0] = arg1;
-              arglist[1] = arg2;
-              arglist[2] = arg3;
-              method = CallHandleMethodNotFound(oper, 3, arglist, 0, 1, prec);
-            }
-          
-          /* update the cache */
-          if (method && prec < INTOBJ_INT(CACHE_SIZE))
-            {
-              Bag cacheBag = GET_METHOD_CACHE( oper, 3 );
-              
```

Comment:
Since we are reformatting: perhaps remove those trailing spaces inside the comments (keeping the "correct" is a PITA, and serves no purpose), or even change it to a `//` comment. The same remarks applies elsewhere (I think `clang-format` would also do it, doesn't it?)